### DISCUSSION TITLE
Handling Multipart form data with the full filename set as name in Content-Disposition

### DISCUSSION THREAD

**Discussion by mntolia** [fastapi/fastapi#6735] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import List

from fastapi import FastAPI, Request, File, UploadFile

app = FastAPI()


@app.post("/1")
async def create_upload_files(files: List[UploadFile] = File(...)):
    return {"filenames": [file.filename for file in files]}
```

### Description

I am getting 2 files as form data from an IoT device. One is a XML file and the other is an image file
The issue is that the "name" part of the Content-Disposition is set as "anpr.xml" and "licensePlatePicture.jpg" for each file respectively. 
I am not sure how to parse this in FastAPI as I can't set anpr.xml as the variable name for the method.

This is the request header:
```
Headers({'accept': 'text/html, application/xhtml+xml, */*', 'accept-language': 'zh-CN', 'content-type': 'multipart/form-data;boundary=-------------------------7e13971310878', 'user-agent': 'Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; WOW64; Trident/5.0)', 'accept-encoding': 'gzip, deflate', 'host': '****', 'content-length': '419354', 'connection': 'Keep-Alive', 'cache-control': 'no-cache'})
```

This is a redacted request body:

```
---------------------------7e13971310878
Content-Disposition: form-data; name="anpr.xml";filename="anpr.xml";
Content-Type: application/xml; charset="UTF-8"
Content-Length: 2672

<?xml version="1.0" encoding="utf-8"?>
.....
---------------------------7e13971310878
Content-Disposition: form-data;name="licensePlatePicture.jpg";filename="licensePlatePicture.jpg";
Content-Type: image/pjpeg
Content-Length: 2124
......
---------------------------7e13971310878--
```

How do I process this in FastAPI?

**Comment by nymous** [comment, marked as answer]
Hi!

You had an answer somewhere else but I'll add it here for the record ^^

You can [specify aliases](https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation) to Pydantic fields, which allows you to give them name that are not valid Python identifiers. For example:
```py
@app.post("/1")
async def create_upload_files(anpr: UploadFile = File(..., alias="anpr.xml"), license_plate: UploadFile = File(..., alias="licensePlatePicture.jpg")):
    return {"filenames": [anpr.filename, license_plate.filename]}
```

**Comment by mntolia** [comment]
This resolves the issue. Thanks @nymous .

**Comment by tiangolo** [comment]
Thanks for the help here @nymous ! 👏  🙇 

Thanks for reporting back and closing the issue @mntolia  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
